homeassistant/components/zwave_js/discovery.py

```diff
@@ -1235,15 +1235,21 @@ def async_discover_single_value(
             continue
 
         # check device_class_generic
-        if value.node.device_class and not check_device_class(
-            value.node.device_class.generic, schema.device_class_generic
-        ):
+        if (
+            value.node.device_class
+            and not check_device_class(
+                value.node.device_class.generic, schema.device_class_generic
+            )
+        ) or (not value.node.device_class and schema.device_class_generic):
```

Comment:
I think we can simplify this check. We know that `check_device_class` will always return true if the required value is `None`. So in both of these cases the generic device class will need to be `not None` for the check to return True.

```py
if (
    schema.device_class_generic
    and (
        not value.node.device_class or not check_device_class(
            value.node.device_class.generic, schema.device_class_generic
        )
    )
):
```

I'd also consider making the required value in `check_device_class` non optional since it will always return True for a None value. We don't need a function for that check.